### ISSUE TITLE
Barb doesn't use item find on Threshsocket

### ISSUE THREAD

**Issue by PeterKrush** [kolton/d2bot-with-kolbot#1177]
Been trying to mess with the Barb attack script, but to no avail. It seems d2bs recognizes threshsocket as a boss, because it will mfweapon swap at low life %. Anyone able to help me out? Thank you

**Comment by DarkHorseDre** [comment]
check `barbarian.js` - corpse checks are done in there, with limitations

**Comment by PeterKrush** [comment]
So this is what i've found on item find in barbarian.js.


`	findItem: function (range) {
		if (!Config.FindItem || !me.getSkill(142, 1)) {
			return false;
		}

		var i, j, tick, corpse, orgX, orgY, retry,
			corpseList = [];

		orgX = me.x;
		orgY = me.y;

MainLoop:
		for (i = 0; i < 3; i += 1) {
			corpse = getUnit(1);

			if (corpse) {
				do {
					if ((corpse.mode === 0 || corpse.mode === 12) && getDistance(corpse, orgX, orgY) <= range && this.checkCorpse(corpse)) {
						corpseList.push(copyUnit(corpse));
					}
				} while (corpse.getNext());
			}

			while (corpseList.length > 0) {
				if (this.checkCloseMonsters(5)) {
					if (Config.FindItemSwitch) {
						Precast.weaponSwitch(Math.abs(Config.FindItemSwitch - 1));
					}

					Attack.clear(10, false, false, false, false);

					retry = true;

					break MainLoop;
				}

				corpseList.sort(Sort.units);

				corpse = corpseList.shift();

				if (this.checkCorpse(corpse)) {
					if (getDistance(me, corpse) > 30 || checkCollision(me, corpse, 0x1)) {
						Pather.moveToUnit(corpse);
					}

					if (Config.FindItemSwitch) {
						Precast.weaponSwitch(Config.FindItemSwitch);
					}
`

I've tried a few things, like putting thresh's ID, from "SuperUniquePresetIDs", into corpseList = [];.
Tried a few other things but could someone point me in the right direction?
ty

**Comment by PeterKrush** [comment]
Is there a corpse list somewhere i can edit?

**Comment by DarkHorseDre** [comment]
is this a barb running as leader or as mfhelper?  

The only restrictions I see are:

if (getDistance(me, unit) <= 25 &&
				!unit.getState(1) && // freeze
				!unit.getState(96) && // revive
				!unit.getState(99) && // redeemed
				!unit.getState(104) && // nodraw
				!unit.getState(107) && // shatter
				!unit.getState(118) // noselect

so as long as the corpse is valid, should be ok - can you finditem manually? I assume so..

if this is the barb as leader/mfleader I'd try adding the code to finditem within the `threshsocket.js` - if all the above are fine then it should work.  exception is the references to `monstats2` and the `unit.classid <= 575` which I don't have time to workout right now..  

Do you know the classid of threshsocket?

**Comment by PeterKrush** [comment]
> is this a barb running as leader or as mfhelper?
> 
> The only restrictions I see are:
> 
> if (getDistance(me, unit) <= 25 &&
> !unit.getState(1) && // freeze
> !unit.getState(96) && // revive
> !unit.getState(99) && // redeemed
> !unit.getState(104) && // nodraw
> !unit.getState(107) && // shatter
> !unit.getState(118) // noselect
> 
> so as long as the corpse is valid, should be ok - can you finditem manually? I assume so..
> 
> if this is the barb as leader/mfleader I'd try adding the code to finditem within the `threshsocket.js` - if all the above are fine then it should work. exception is the references to `monstats2` and the `unit.classid <= 575` which I don't have time to workout right now..
> 
> Do you know the classid of threshsocket?

I'm running him with leader script, yes i can pop his body manually.
"755	Threash Socket"  - From Superunique_presetunitids.txt

Is the class Id different than the unit id?
Also noticed he doesn't use find item at Nihalthek
gonna mess around with threshsocket.js a bit thanks for your help

**Comment by DarkHorseDre** [comment]
ok that 755 looks like the classid its looking for,
So as long as the item/skill you use to kill him doesn't end with it being in one of the listed states or distances it should be cool...

I'd force it in threshsocket.js if you cant adjust it in the barb file. this way, you cast finditem without worrying about the classid (unless the game is hardcoded to ignore him?) you'll soon see..

interestingly, superuniques.txt spells him as ```Threash Socket	51```
superunique_presetunitids.txt spell it as ```755	Threash Socket``` - consistent at least! not sure if it is throughout the game tho?

**Comment by PeterKrush** [comment]
> ok that 755 looks like the classid its looking for,
> So as long as the item/skill you use to kill him doesn't end with it being in one of the listed states or distances it should be cool...
> 
> I'd force it in threshsocket.js if you cant adjust it in the barb file. this way, you cast finditem without worrying about the classid (unless the game is hardcoded to ignore him?) you'll soon see..
> 
> interestingly, superuniques.txt spells him as `Threash Socket	51`
> superunique_presetunitids.txt spell it as `755	Threash Socket` - consistent at least! not sure if it is throughout the game tho?

yeah that bit threw me for a loop lol wasn't sure about his ID. but i edited my post above, got it working, thanks!

**Comment by DarkHorseDre** [comment]
Ha! I was gonna point out the string id but didn't think it was relevant!

so kill vs clear - how/why the hell does that work?!?

did you put that fix in the threshsocket or barb script ?

**Comment by PeterKrush** [comment]
Just in threshsocket.js

It seems that the kill function is used on bosses, and there's some restriction somewhere on using find item on bosses. Set it to clear and it reads thresh as a champion rather than a boss.

This is just a guess tbh

**Comment by PeterKrush** [comment]
Couldn't figure out how to add that in correctly after killing thresh,, my knowledge is very limited!

**Comment by DarkHorseDre** [comment]
Mine too :)
